src/main/java/net/dv8tion/jda/core/audio/AudioWebSocket.java

```diff
@@ -521,7 +517,7 @@ private void setupKeepAlive(final int keepAliveInterval)
 
         try
         {
-            keepAlivePool.scheduleAtFixedRate(keepAliveRunnable, 0, keepAliveInterval, TimeUnit.MILLISECONDS);
+            keepAliveHandle = keepAlivePool.scheduleAtFixedRate(keepAliveRunnable, 0, keepAliveInterval, TimeUnit.MILLISECONDS);
```

Comment:
Hmm okay, but it doesn't recover still, and we'll have a dangling runnable, which was the problem in the first place.
Just letting you know.
But if you think it's fine disregard my comment.